File: odemapster-core/src/main/java/es/upm/fi/dia/oeg/obdi/core/engine/IQueryTranslator.java
package es.upm.fi.dia.oeg.obdi.core.engine;

import java.sql.Connection;

import com.hp.hpl.jena.graph.Triple;
import com.hp.hpl.jena.query.Query;

import es.upm.fi.dia.oeg.morph.base.ConfigurationProperties;
import es.upm.fi.dia.oeg.morph.base.TermMapResult;
import es.upm.fi.dia.oeg.obdi.core.model.AbstractMappingDocument;
import es.upm.fi.dia.oeg.obdi.core.sql.IQuery;

public interface IQueryTranslator {
	Connection getConnection();
	
	void setOptimizer(IQueryTranslationOptimizer optimizer);

	Query getSPARQLQuery();
	
	void setSPARQLQuery(Query query);
	
	void setSPARQLQueryByString(String queryString);
	
	void setSPARQLQueryByFile(String queryFilePath);
	
	IQuery getTranslationResult();
	
	IQuery translate(Query query) throws Exception;

	void setMappingDocument(AbstractMappingDocument md);
	
	AbstractMappingDocument getMappingDocument();

	//void setUnfolder(AbstractUnfolder unfolder);

	//void setIgnoreRDFTypeStatement(boolean b);

	IQuery translateFromQueryFile(String queryFilePath) throws Exception;

	//IQueryTranslationOptimizer getOptimizer();

	IQuery translateFromString(String queryString) throws Exception ;

	//String translateResultSet(String columnLabel, String dbValue);
	
	TermMapResult translateResultSet(String varName, AbstractResultSet rs);
	
	void setConfigurationProperties(ConfigurationProperties configurationProperties);

	void setDatabaseType(String databaseType);
	
	String getDatabaseType();

	//void setConnection(Connection conn);
	
//	String getTripleAlias(Triple triple);
	
	public abstract AbstractUnfolder getUnfolder();
	
//	public abstract void putMappedMapping(Integer key, Object value);
	
//	public abstract void putTripleAlias(Triple tp, String alias);

}

File: morph-base/src/main/scala/es/upm/fi/dia/oeg/morph/base/engine/IQueryTranslator.scala
package es.upm.fi.dia.oeg.morph.base.engine

import java.sql.{Connection, ResultSet}

import org.apache.jena.query.Query
import es.upm.fi.dia.oeg.morph.base.sql.IQuery
import es.upm.fi.dia.oeg.morph.base.model.MorphBaseMappingDocument
import es.upm.fi.dia.oeg.morph.base.{MorphProperties, TranslatedValue}
import org.apache.jena.graph.Node
//import com.hp.hpl.jena.sparql.algebra.Op
import org.apache.jena.sparql.algebra.Op;
//import com.hp.hpl.jena.sparql.algebra.op.OpBGP
import org.apache.jena.sparql.algebra.op.OpBGP;
import Zql.ZUpdate
import Zql.ZInsert
import Zql.ZDelete

trait IQueryTranslator {
  var connection:Connection = null;

  //	var sparqlQuery :Query = null;

  var optimizer:QueryTranslationOptimizer  = null;

  var properties:MorphProperties =null;

  var databaseType:String =null;

  //val unfolder:MorphBaseUnfolder=null;

  var mappingDocument:MorphBaseMappingDocument= null;

  //	def setSPARQLQueryByString(queryString:String );
  //
  //	def setSPARQLQueryByFile(queryFilePath:String );

  //def getTranslationResult():IQuery ;

  def translate(query:Query) :IQuery ;

  def translate(op:Op) :IQuery ;

  def translateFromQueryFile(queryFilePath:String ) : IQuery;

  //IQueryTranslationOptimizer getOptimizer();

  def translateFromString(queryString:String ) : IQuery;

  //String translateResultSet(String columnLabel, String dbValue);

  def generateNode(rs:ResultSet, varName:String, mapXSDDatatype:Map[String, String]
                   , varNameColumnLabels:List[String]):Node;

  def setDatabaseType(dbType:String) = {this.databaseType = dbType}

  def trans(op:Op) : IQuery;

  def translateUpdate(stg:OpBGP) : ZUpdate;

  def translateInsert(stg:OpBGP) : ZInsert;

  def translateDelete(stg:OpBGP) : ZDelete;

  def setConnection(conn:Connection) = {this.connection = conn}

  def setOptimizer(optimizer:QueryTranslationOptimizer) = { this.optimizer = optimizer }
}